lib/github-pages-health-check/domain.rb

```diff
@@ -199,7 +203,8 @@ def a_record?
       # Is this domain's first response a CNAME record?
       def cname_record?
         return unless dns?
-        dns.first.class == Net::DNS::RR::CNAME
+        dns.first.class == Net::DNS::RR::CNAME &&
+          Domain.valid_domain?(dns.first.cname.to_s)
```

Comment:
I believe this can be `cname.valid_domain?` We already instantiate a domain for the cname via https://github.com/github/pages-health-check/blob/master/lib/github-pages-health-check/domain.rb#L210. 

Perhaps it'd make more sense to have the CNAME method confirm the class, and then `cname?` can just look for the existence?